### PR TITLE
Upgrade to Smallrye Fault Tolerance 6.5.0 and resolve FT methods during build

### PR DESCRIPTION
**Pull request by Ladicek**
- https://smallrye.io/blog/fault-tolerance-6-5-0/

### CODE REVIEW

**Comment by Ladicek** [comment]
This builds on a new feature of SmallRye Fault Tolerance 6.5.0, which requires fault tolerance methods to be resolved during deployment -- which means at build time in Quarkus. This is what most of the changes in this PR relate to.

CC @zakkak -- this is what we discussed a while ago (see also #42229).

**Review by gsmet — APPROVED** [review]
This looks good to me. I just want to check with @radcortez that we are OK partially updating to next MicroProfile in 3.16 (because my understanding is that it's what we do here, correct me if I'm mistaken).

**Comment by Ladicek** [comment]
Yeah, that's right. MP FT 4.1 is part of MicroProfile 7.0.

I don't think we've coordinated updates in the past, but if @radcortez wants, this can certainly wait.

File: extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/FaultToleranceMethodSearch.java [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.TypeVariable;
+import org.jboss.jandex.VoidType;
+import org.jboss.jandex.WildcardType;
+
+import io.quarkus.arc.processor.AssignabilityCheck;
+import io.quarkus.arc.processor.KotlinDotNames;
+
+// copy of `io.smallrye.faulttolerance.config.SecurityActions` and translation from reflection to Jandex
```
Comment (zakkak) [inline extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/FaultToleranceMethodSearch.java:28]: Don't we need to include the copyright notice from https://github.com/smallrye/smallrye-fault-tolerance/blob/d85d79e02ed704945dce1ddf1dae38834733e89e/implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/config/SecurityActions.java#L1-L15 as discussed in https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Reusing.20code.20from.20different.20project.20under.20apache.20license?

Comment (Ladicek) [reply extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/FaultToleranceMethodSearch.java:28]: That file is authored by myself and @michalszynkiewicz in the past, when he was at Red Hat, so I didn't think this matters, but I'll add a copy of the header here just to be sure.

Comment (Ladicek) [reply extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/FaultToleranceMethodSearch.java:28]: Done. Thanks!

**Comment by radcortez** [comment]
> This looks good to me. I just want to check with @radcortez that we are OK partially updating to next MicroProfile in 3.16 (because my understanding is that it's what we do here, correct me if I'm mistaken).

Yes, it is ok. Currently, we stopped being compatible some versions ago with the OTel updates, so it's not a big deal. We do have to ensure that we are able to sync all the work in one of our releases so we can certify again.

**Review by radcortez — APPROVED** [review]

**Comment by gsmet** [comment]
CI failures were unrelated.

**Pull request merged**